zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/EngineProcessors.java

```diff
@@ -586,14 +585,10 @@ private static void addCommandDistributionProcessors(
       final CommandDistributionBehavior commandDistributionBehavior,
       final TypedRecordProcessors typedRecordProcessors,
       final Writers writers,
-      final ProcessingState processingState,
-      final Supplier<ScheduledTaskState> scheduledTaskStateFactory,
-      final InterPartitionCommandSender interPartitionCommandSender) {
+      final ProcessingState processingState) {
 
     // periodically retries command distribution
-    typedRecordProcessors.withListener(
-        new CommandRedistributor(
-            scheduledTaskStateFactory.get().getDistributionState(), interPartitionCommandSender));
```

Comment:
@korthout this is the change I am most unsure about 🤔

Reply:
Yes, this is not safe IMO. It's important that the `CommandRedistributor` uses the state provided by `scheduledTaskStateFactory.get()`.

Reply:
Yeah that's what I thought as well. Thus if I want to follow the pattern to use the behavior in the redistributor, then I would need to create a new behavior, right?